File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.PriorityQueue;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        PriorityQueue<Integer> pq = new PriorityQueue<>();

        pq.add(score[0]);
        answer[0] = score[0];
        for(int i = 1; i<score.length; i++){
            if(pq.size() == k && pq.peek() <= score[i]){
                pq.poll();
                pq.add(score[i]);
            }else if(pq.size() < k){
                pq.add(score[i]);
            }

            answer[i] = pq.peek();
        }


        return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    let answer = [];
    let result = [];

    for(x of score){

        answer.push(x);
        answer = answer.sort((a,b) => (b-a)).splice(0,k);

        result.push(answer[answer.length-1]);
    }
    return result;
}